ARCHITECTURE.md

```
@@ -0,0 +1,190 @@
+   [`workspace/positron/libpositron_cuda.nim`](workspace/positron/libpositron_cuda.nim).
+
+7. **`crucible`** ([`workspace/crucible/crucible.nim`](workspace/crucible/crucible.nim))
+   is the GPU kernel code generator: Nim macros consume ceramic layout algebra
+   and positron kernel specs and emit native CUDA, OpenCL, Vulkan and WebGPU
+   code ([`workspace/crucible/src/codegen/gpu_compiler.nim`](workspace/crucible/src/codegen/gpu_compiler.nim),
+   targets under [`workspace/crucible/src/codegen/targets/`](workspace/crucible/src/codegen/targets/),
+   ABIs under [`workspace/crucible/src/abis/`](workspace/crucible/src/abis/)).
+
+## Repo layout
+
+```
```

markdownlint-cli2 (0.23.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)